python/lsst/ts/FiberSpectrograph/dataManager.py

```diff
@@ -0,0 +1,130 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["SpectrographData", "DataManager"]
+
+import dataclasses
+import os.path
+
+import astropy.time
+import astropy.io.fits
+import numpy as np
+
+
+@dataclasses.dataclass
+class SpectrographData:
+    """Class to hold data and metadata from a fiber spectrograph.
+    """
+    wavelength: np.ndarray
+    """The wavelength array produced by the instrument in nm."""
+    spectrum: np.ndarray
+    """The flux array in instrumental units."""
+    duration: float
+    """The duration of the exposure in seconds."""
+    date_begin: astropy.time
+    """The start time of the exposure (TAI)."""
+    date_end: astropy.time
+    """The end time of the exposure (TAI)."""
+    type: str
+    """The measurement type (see the XML `expose.type` value)."""
+    source: str
+    """The light source that was measured (see the XML `expose.source` field).
+    """
+    temperature: float
+    """The internal spectrograph temperature."""
+    temperature_setpoint: float
+    """The internal spectrograph temperature set point."""
+    n_pixels: int
+    """The number of pixels in the detector."""
+
+
+class DataManager:
+    """A data packager and uploader for output from the Fiber Spectrograph CSC.
+
+    Attributes
+    ----------
+    telescope : `str`
+        The name of the telescope the instrument is attached to.
+    instrument : `str`
+        The name of the instrument taking the data.
+    outpath : `str`, optional
+        A path to write the FITS files to.
+        **WARNING**: this argument will be removed once the API for writing
+        to the LFA becomes available in salobj; do not rely on it.
+    """
+    def __init__(self, telescope, instrument, outpath=None):
+        self.telescope = telescope
+        self.instrument = instrument
+        self.outpath = outpath
+
+    def __call__(self, data):
+        """Generate a FITS hdulist built from SpectrographData.
+
+        Parameters
+        ----------
+        data : `SpectrographData`
+            The data to build the FITS output with.
+
+        Returns
+        -------
+        uri : `str`
+            The file path or other identifier (e.g. s3 store) for where the
+            data was sent.
+        """
+        hdu1 = self.make_primary_hdu(data)
+        hdu2 = self.make_wavelength_hdu(data)
+        hdulist = astropy.io.fits.HDUList([hdu1, hdu2])
+        # TODO: this writeto(file) block should be removed once we have a
+        # working LFA API to send data to.
+        if self.outpath is not None:
+            filename = os.path.join(self.outpath, f"{self.instrument}_{data.date_begin.tai.fits}.fits")
+            hdulist.writeto(filename, checksum=True)
+            return filename
+        else:
+            return None
+
+    def make_fits_header(self, data):
+        """Return a FITS header built from SpectrographData."""
```

Comment:
Said python dict would already have FITS keys baked into it though; why else would you name a dictionary key `INSTRUME` or `EXPTIME` instead of `instrument` and `exposure_time`? That seems like a distinction without a difference.

Reply:
If you start dealing with comment strings and units then it clearly makes sense to work with FITS objects here (or PropertyList if you were in DM).